### DISCUSSION TITLE
Unable to convert Deposits / Withdrawals to Transfers

### DISCUSSION THREAD

**Discussion by MrBogger** [firefly-iii/firefly-iii#9068] [Firefly III]
Good morning, 
Firstly, thanks to everyone who has contributed to this project. It's a great product.
Secondly, I'm trying to clean up a lot of my imported transactions and I've been having trouble understanding the limitations around transfers between accounts. 
I am trying to reconcile a lot of the transactions by changing the type to a transfer but when I am trying to go back and forth with liabilities and assets, it won't let me select the correct account. I understand there is normally one way flows of money to a liability and I have read and understood the Docs on the transaction types but there are occasions where I have had a transaction reversed or I have transferred excess money from the loan account to an asset account. Is there a way of 'unlocking' the transaction conditions on the accounts or something I need to be doing differently so I am able to achieve this? This is the same when using Deposit / Withdrawal transactions between accounts instead of transfers where I am unable to select the asset account from a liability account and vice versa.
At the moment, I have 3 accounts listed under liabilities and my bank allows me to transfer excess money into any of the loans and then withdraw it out as I need to.

Thanks,

**Comment by JC5** [comment]
<img width="919" alt="Screenshot 2024-07-21 at 17 13 48" src="https://github.com/user-attachments/assets/6e76593b-b95c-45d4-ad25-0964936cc42c">

Use the buttons in the drop down to convert deposits and withdrawals to transfers. If transfers need to go the other way around, all you have to do is edit them, and change the source + destination.

**Comment by MrBogger** [comment]
Hey JC5,
 
Thanks for the reply. I've worked out I have two related issues when trying to move money between Asset accounts and Liability Accounts.

Firstly, I am only able to have transfer transactions between Asset accounts only, or Liability accounts only. Transactions between Assets and Liabilities can only be withdrawal/deposits. I am ok with this method but would prefer it to be a transfer to differentiate easier from normal withdrawals/deposits as the money is still mine in the sense and it hasn't gone to someone else until the bank takes it as another withdrawal.

Secondly, when trying to sort through all of my imported transactions, if the transaction had the (no name) tag as an Expense or Revenue account, there are restrictions on which accounts I can select to correct the transaction. For example, if I have a transaction which has been allocated as a withdrawal and has (no name) in the destination, I am unable to edit this to have an Asset account destination (either Expense or Liability accounts only). I then need to create a new transaction and enter all the details and Cloning doesn't work for this either. If there is a transaction that's been allocated as Revenue, I am only able to change the source account to either Asset or Revenue even if the money came from a Liability account. Again needing to create a new blank transaction. 

I understand this is a downfall of the importer not recognising and matching transfer transactions between bank accounts especially when the description is different in each account. I had selected everything to be mapped during import and I had left my main account until last  to try and have the best shot of reducing the clutter. I had created rules to correct my transactions and transfers but they just created more expense and revenue accounts instead of the correct asset and liability accounts.

Cheers,
Russell

**Reply by JC5** [reply]
> Firstly, I am only able to have transfer transactions between Asset accounts only, or Liability accounts only. Transactions between Assets and Liabilities can only be withdrawal/deposits. I am ok with this method but would prefer it to be a transfer to differentiate easier from normal withdrawals/deposits as the money is still mine in the sense and it hasn't gone to someone else until the bank takes it as another withdrawal.

Yes, this is by design. Sorry.

> Secondly, when trying to sort through all of my imported transactions, if the transaction had the (no name) tag as an Expense or Revenue account, there are restrictions on which accounts I can select to correct the transaction. 

Please check the screenshot to learn how to correct them. You must convert them, see the dropdown.

>  I had created rules to correct my transactions and transfers but they just created more expense and revenue accounts instead of the correct asset and liability accounts.

Select the rule action "convert to X": https://docs.firefly-iii.org/references/firefly-iii/rule-actions/#converting-to-another-transaction-type

**Comment by MrBogger** [comment]
Hey James,
Please forgive me if I seem hopelessly lost and trying to understand the layout and transaction types etc. I'm working it all out eventually.

So, from what I've gathered, any transactions from an Asset account to a Liability MUST be a withdrawal for it to be able to select the correct accounts. Same goes for withdrawing money from the Liability back into the Asset account. I was under the impression that I could create a deposit to the Asset account from the Liability account and Firefly would display the transaction in both accounts like what the withdrawal does. I understood the Deposit Transaction to be the opposite of a withdrawal rather than only recording money coming in from external sources. Also, because I have all my accounts with the same bank, all of my transactions between Assets and Liabilities are internal transfers. So in the end, I'll need to remove all of my deposit transactions between my accounts and only keep the withdrawals.

Regarding my issue / bug report, on the liability account transaction list page, when a withdrawal is made from an Asset account into a Liability account, should the amount then show up as an incoming deposit rather than a withdrawal still? As it is calculated correctly in the Grand Total being that incoming money will bring a negative account closer to zero and any interest/fees/withdrawals taken out will make it less/away from $0.
Example is one of my accounts below. My understanding is the monthly box on the side should show an incoming amount of $3068 and an outgoing amount of $2650 which is what the line graph shows and displayed by the grand total. It should be a net balance difference of +$418 rather than a calculated amount of -$5718 leaving the account. Please feel free to delete or remove the issue that I raised.
<img width="1186" alt="Screen Shot 2024-07-22 at 21 22 33" src="https://github.com/user-attachments/assets/b7f542b0-0daf-44f6-b610-a19f27ba677c">

**Reply by JC5** [reply]
This is something to consider for the new layout :)
